### PR TITLE
wine: add x86_64 target with i686 support

### PR DESCRIPTION
**Pull request by tim241**
I'm sorry about the closure of the [previous PR](https://github.com/void-linux/void-packages/pull/350), I made a mistake and I couldn't restore the branch due to my stupidity, won't happen again.

### CODE REVIEW

File: srcpkgs/wine/template [hunk 1]
```diff
@@ -40,20 +88,25 @@ post_install() {
...
 	}
 }
 wine-devel_package() {
 	depends="libwine-${version}_${revision}"
 	short_desc+=" - development files"
 	replaces="wine-unstable-devel>=0"
 	pkg_install() {
-		vmove usr/include
-		vmove "usr/lib/*.so"
+		vmove "usr/include"
```
Comment (maxice8) [inline srcpkgs/wine/template]: don't need to quote a fixed string path, only globs are required.

**Comment by abenson** [comment]
If you can update this to wine 3.12, that'd be great.

Using this in conjunction with the expected libGL 18.x updates, DXVK, etc, and it all works great!

**Comment by nathanielhorn** [comment]
I've tested this version as well and it build succesfully including dxvk/vulkan support. everything runs great. hope this gets merged soon

**Comment by abenson** [comment]
Needs a bump to 3.13, but I think it's great

**Comment by tim241** [comment]
@abenson there's already a PR to bump the wine version: https://github.com/void-linux/void-packages/pull/1206

**Comment by abenson** [comment]
But wine64..

**Comment by tim241** [comment]
@abenson wine64 doesn't need to be bumped since it's from the same source as win32.

**Comment by abenson** [comment]
Well, when that PR gets merged, this one will conflict and you'll have to rework it anyway.

**Comment by tim241** [comment]
@abenson it isn't much work for me to "rework" this PR
(I probably do this wrong but it works)
I clone this repo(void-linux/void-packages)
then I apply my patch(with version number modified)
git commit (with my message)
git checkout -b [branch]
then I do a force push to my fork and the PR will be updated

**Comment by Sacules** [comment]
I'm very interested in this, hope it gets merged soon :smiley:

**Comment by tim241** [comment]
I hope it'll be merged soon too!

**Comment by Yardanico** [comment]
any status on this? :(

**Comment by tim241** [comment]
I have no idea, I updated the PR to solve the conflict(thanks for
reminding me!)

Other than that, no word from the maintainer(s) yet sadly

**Comment by Hoshpak** [comment]
I'm afraid this can't be merged because `xbps-src` does not support what you are trying to do right now. Currently the only option seems to be to create separate wine and wine64 packages, thus losing the `WoW64` functionality. Currently I'm experimenting a bit with `xbps-src` to be able to give you a substantiated explanation of why it most likely won't be accepted. (Or possibly prove my assumptions wrong, we will see)

**Comment by Hoshpak** [comment]
So my computer just finished compiling all the dependencies, I will begin from the start so that everyone gets the full picture:

Void never had true multiarch support the way Debian or other distributions implemented it. The only thing that comes somewhat close to it is the building of the `-32bit` packages  which is more of a hack. It uses the i686 binaries built by the i686 builder and repackages them as `x86_64` binary packages with the `-32bit' suffix. So building the wine package the way proposed in this PR would require the necessary builds to have run on the i686 builder first and the multilib repository to be enabled during the build. Since both is the case in the current configuration, that would work if the template hits the builders.

However `void-packages` is built on the premise that is is possible to bootstrap everything from sources without requiring any preexisting binary packages. So I should be able to always just clone this repository, not use any of the mirrors and build a whole system from the sources. That's also the reason why packages that require themselves to bootstrap something have two versions of them in the repository, one being binaries provided by upstream which can be used to bootstrap and build the package from source, the other one being the source of the packages which can the be built. Now if I try to compile wine with this template without using the existing repositories, it will fail because `xbps-src` can't find any of the `-32bit` packages. There is now template for them and `xbps-src` can't detect that it should build the packages without suffix for i686 first to have them available. That's what was refered to in https://github.com/voidlinux/void-packages/pull/11534 as  
> bootstrap issues

The proper solution for this would be to implement true multiarch support in Void which would be a huge amount of work and extend `xbps-src` to make it possible to depend on packages for a different architecture and then automatically (cross)compiling the necessary packages for the specified architecture.

**Comment by tim241** [comment]
@Vintodrimmer already did that! https://github.com/tim241/void-packages/tree/wine-staging-wip 

just change the version number & hashes to compile the version you want :+1:

**Comment by youngjinpark20** [comment]
@tim241 can you edit the changes for new wine template

**Comment by ang9711** [comment]
3

**Comment by ang9711** [comment]
#540

**Comment by youngjinpark20** [comment]
@tim241 thanks for your good work :1st_place_medal:

**Comment by Vintodrimmer** [comment]
Ping @maxice8 .

Is there anything still stopping this from getting into the repos?

**Comment by tim241** [comment]
@Vintodrimmer 
> Is there anything still stopping this from getting into the repos?

yes, read https://github.com/void-linux/void-packages/pull/540#issuecomment-417840507 and https://github.com/void-linux/void-packages/pull/540#issuecomment-417855681

**Comment by maxice8** [comment]
On Sun, Feb 03, 2019 at 07:39:13AM -0800, Vintodrimmer wrote:
> Ping @maxice8 .
> 
> Is there anything still stopping this from getting into the repos?
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/540#issuecomment-460062161

Don't ask me, i have no interest in this.

**Comment by Gottox** [comment]
superseded by #11895

**Comment by tim241** [comment]
@Gottox Why? This PR (and yours presumably) is never gonna be merged due to https://github.com/void-linux/void-packages/pull/540#issuecomment-417840507 and https://github.com/void-linux/void-packages/pull/540#issuecomment-417855681 ?

**Pull request closed without merging**